src/main/java/com/google/api/codegen/transformer/go/GoSurfaceNamer.java

```diff
@@ -94,4 +109,89 @@ private static String upperFirstLetter(String s) {
   public String getFieldGetFunctionName(TypeRef type, Name identifier) {
     return methodName(identifier);
   }
+
+  public String getGrpcClientTypeName() {
+    // getNickname() thinks the client is a message type and appends a "*".
+    // substring gets rid of the star.
+    return converter.getTypeName(service).getNickname().substring(1) + "Client";
+  }
+
+  public String getGrpcClientConstructorName() {
+    return getGrpcClientTypeName().replace(".", ".New");
+  }
+
+  public String getCallOptionsName() {
+    return clientNamePrefix() + "CallOptions";
+  }
+
+  public String getDefaultClientOptionFunc() {
+    return "default" + getClientName() + "Options";
+  }
+
+  public String getDefaultCallOptionFunc() {
+    return "default" + getCallOptionsName();
+  }
+
+  public String getClientName() {
+    return clientNamePrefix() + "Client";
+  }
+
+  public String getClientConstructorName() {
+    return "New" + getClientName();
```

Comment:
You have a bunch of "unsafe" concatenations in the above functions, in the sense that you don't know from this context what the casing of getClientName() is (e.g. if it is overridden or if it changes elsewhere). Use the Name class for concatenation because it ensures everything is concatenated together correctly in a casing-aware way.